File: src/main/java/com/easy/zssn/model/Survivor.java
package com.easy.zssn.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class Survivor{
    @Id
    @GeneratedValue
    private Integer id;
    private String name;
    private int age=0;
    private String gender;
    private boolean isInfected=false;
    private int reports=0;

    public void addReport(){
        reports++;
        if(reports>=3)
            isInfected=true;
    }

    public boolean isInfected() {
        return isInfected;
    }
    public void setInfected(boolean isInfected) {
        this.isInfected = isInfected;
    }
    public Integer getId() {
        return id;
    }
    public void setId(Integer id) {
        this.id = id;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }
    public int getAge() {
        return age;
    }
    public void setAge(int age) {
        this.age = age;
    }
    public String getGender() {
        return gender;
    }
    public void setGender(String gender) {
        this.gender = gender;
    }
    
    @Override
    public String toString(){
        return "id: "+this.id+"\n"+
        "name: "+this.name+"\n"+
        "age: "+this.age+"\n"+
        "gender: "+this.gender+"\n"+
        "isInfected: "+this.isInfected+"\n"+
        "reports: "+this.reports+"\n";
    }
}

File: src/main/kotlin/com/eduardo/rdguez/zssn/domain/Survivor.kt
package com.eduardo.rdguez.zssn.domain

import com.eduardo.rdguez.zssn.model.enums.Gender
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.EnumType
import javax.persistence.Enumerated
import javax.persistence.FetchType
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.OneToMany
import javax.persistence.OneToOne
import javax.persistence.Table
import org.hibernate.annotations.Fetch
import org.hibernate.annotations.FetchMode

@Entity
@Table(name = "survivor")
class Survivor(
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "id", nullable = false)
  var id: Long? = null,

  @Column(name = "name", nullable = false)
  var name: String,

  @Column(name = "age", nullable = false)
  var age: Int,

  @Column(name = "gender", nullable = false)
  @Enumerated(EnumType.STRING)
  var gender: Gender,

  @Column(name = "is_infected", nullable = true)
  var isInfected: Boolean? = false,

  @OneToOne(fetch = FetchType.LAZY)
  @Fetch(FetchMode.JOIN)
  @JoinColumn(name = "location_id", nullable = false)
  var lastLocation: Location,

  @OneToMany(fetch = FetchType.LAZY)
  @Fetch(FetchMode.JOIN)
  @JoinColumn(name = "survivor_id")
  var survivorInventory: List<SurvivorInventory> = listOf(),
)